### PR TITLE
Do not build with internal SQLite

### LINKED ISSUE #105: Problem with compilation
When I try to compile geodiff from source on a clean build directory on ubuntu 20.04, I get the following error during `make`

```
[ 84%] Linking CXX executable geodiff_single_commit_test
/usr/bin/ld: ../libgeodiff_a.a(sqlitedriver.cpp.o): undefined reference to symbol 'sqlite3_column_text'
/usr/bin/ld: /lib/x86_64-linux-gnu/libsqlite3.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/geodiff_single_commit_test.dir/build.make:91: tests/geodiff_single_commit_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:1070: tests/CMakeFiles/geodiff_single_commit_test.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

```

### PR DESCRIPTION
**Pull request by wonder-sk**
We do not need sqlite3 session extension anymore, so let's get rid of the internal copy of sqlite and just use whatever is available on the system.

This should also help us to avoid possible database corruption when pygeodiff in QGIS is using its own sqlite copy and that clashes with the normal copy:
https://github.com/lutraconsulting/qgis-mergin-plugin/issues/225

Fixes #105

### CODE REVIEW

File: geodiff/CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,6 @@ SET(ENABLE_TESTS TRUE CACHE BOOL "Build tests?")
 SET(ENABLE_COVERAGE FALSE CACHE BOOL "Enable GCOV code coverage?")
 SET(BUILD_TOOLS TRUE CACHE BOOL "Build tool executables?")
 SET(BUILD_STATIC FALSE CACHE BOOL "Build static libraries?")
-SET(WITH_INTERNAL_SQLITE3 TRUE CACHE BOOL "Use internal copy of SQLITE3")
```
Comment (PeterPetrik) [inline geodiff/CMakeLists.txt:17]: can you do instead a force ERROR message 
like 

```
if (WITH_INTERNAL_SQLITE3)
  message(ERROR this option was removed in geodiff 1.x due to the ....)
endif()

File: geodiff/CMakeLists.txt [hunk 2]
```diff
@@ -211,24 +182,10 @@ IF (BUILD_TOOLS)
 ENDIF (BUILD_TOOLS)
 
 ADD_DEFINITIONS( -DSQLITE_CORE )
```
Comment (PeterPetrik) [inline geodiff/CMakeLists.txt:184]: is this one required still ?

**Review by PeterPetrik — APPROVED** [review]
👏

File: geodiff/CMakeLists.txt (hunk 1 above)
Comment (wonder-sk) [reply geodiff/CMakeLists.txt:17]: added in e148516

File: geodiff/CMakeLists.txt (hunk 2 above)
Comment (wonder-sk) [reply geodiff/CMakeLists.txt:188]: apparently yes, otherwise the compilation fails (both shared and static lib)

**Comment by wonder-sk** [comment]
TODO:
- mac compilation with sqlite from homebrew
- windows compilation (need to download sqlite - precompiled / chocolatey / vcpkg / conda ?)

**Comment by PeterPetrik** [comment]
let me fix CI in the new PR

**Pull request merged**